### ISSUE TITLE
Add confirmation when quitting the workbench

### ISSUE THREAD

**Issue by CodeSandwich** [microsoft/vscode#235118]
When `workbench.action.quit` is triggered, the entire workbench is immediately nuked, all windows are closed and the state is lost. The default shortcut for this is `ctrl + Q`, which is very easy to press accidentally. It would be great to get a confirmation popup when `workbench.action.quit` is triggered. For example Firefox does that and it's an elegant solution. It won't be annoying because it isn't a part of a normal workflow, it's the last step done only once per session. It will save a lot of frustration from the users who will press `ctrl + Q` by accident.

**Comment by gjsjohnmurray** [comment]
Did you try the `window.confirmBeforeClose` setting?

**Comment by CodeSandwich** [comment]
> Did you try the `window.confirmBeforeClose` setting?

I wasn't aware of this setting and it solves the problem for me 👍 WDYT about making the default something other than `never`? A lot of work can be wasted by accidentally closing a window, which is very easy.

**Comment by CodeSandwich** [comment]
"No" would suffice, but OK, I'll open another issue then.
